src/test/java/tests/configurations/EnvConfigurationTests.java

```diff
@@ -0,0 +1,101 @@
+
+    @Test
+    public void testShouldBePossibleToOverrideTimeoutWithEnvVariable() {
+        long conditionTimeout = CustomAqualityServices.getServiceProvider().getInstance(ITimeoutConfiguration.class).getCondition();
+        assertEquals(conditionTimeout, Long.parseLong(NEW_INT_VALUE), "Condition timeout should be overridden with env variable");
+    }
+
+    @Test
+    public void testShouldBePossibleToOverrideRetryConfigurationWithEnvVariable() {
+        int retryNumber = CustomAqualityServices.getServiceProvider().getInstance(IRetryConfiguration.class).getNumber();
+        assertEquals(retryNumber, Long.parseLong(NEW_INT_VALUE), "Number of retry attempts should be overridden with env variable");
+    }
+
+    @Test
+    public void testShouldBePossibleToOverrideElementCacheStateWithEnvVariable() {
+        boolean isEnabled = CustomAqualityServices.getServiceProvider().getInstance(IElementCacheConfiguration.class).isEnabled();
+        assertTrue(isEnabled, "Element cache state should be overridden with env variable");
+    }
+
+    @Test
+    public void testNumberFormatExceptionShouldBeThrownIfTimeoutIsNotANumber() {
+        System.setProperty(CONDITION_TIMEOUT_KEY, NEW_STRING_VALUE);
+        checkNumberFormatException(() -> CustomAqualityServices.getServiceProvider().getInstance(ITimeoutConfiguration.class).getCommand());
+    }
+
+    @Test
+    public void testNumberFormatExceptionShouldBeThrownIfRetryConfigurationIsNotANumber() {
+        System.setProperty(RETRY_NUMBER_KEY, NEW_STRING_VALUE);
+        checkNumberFormatException(() -> CustomAqualityServices.getServiceProvider().getInstance(IRetryConfiguration.class).getNumber());
+    }
+
+    @Test
+    public void testShouldGetDefaultLanguageIfConfigurationIsAbsent() {
+        System.setProperty(PROFILE_KEY, "empty");
+        System.clearProperty(LANGUAGE_KEY);
+        CustomAqualityServices.initInjector(new TestModule(CustomAqualityServices::getApplication));
+        String language = CustomAqualityServices.getServiceProvider().getInstance(ILoggerConfiguration.class).getLanguage();
+        assertEquals(language, DEFAULT_LANGUAGE, "Current language should be got from logger configuration");
+    }
+
+    @AfterMethod
+    public void after() {
+        System.clearProperty(LANGUAGE_KEY);
+        System.clearProperty(CONDITION_TIMEOUT_KEY);
+        System.clearProperty(RETRY_NUMBER_KEY);
+        System.clearProperty(ELEMENT_CACHE);
+        CustomAqualityServices.initInjector(new AqualityModule<>(CustomAqualityServices::getApplication));
+    }
+
+    private void checkNumberFormatException(Runnable getNumberAction) {
+        try {
+            CustomAqualityServices.initInjector(new TestModule(CustomAqualityServices::getApplication));
+            getNumberAction.run();
+        } catch (Exception e) {
+            Assert.assertSame(e.getCause().getClass(), NumberFormatException.class);
+            Assert.assertTrue(e.getMessage().contains("NumberFormatException"));
```

Comment:
is it necessary after first verification on exception type?

Reply:
I am not sure if nedded second assert. Type of exception should be NumberFormatException